tests/pc/test_retry_count.py

```diff
@@ -0,0 +1,361 @@
+        ByteField("terminator_length", 0),
+        ConditionalField(XStrFixedLenField("reserved", "", 42), lambda pkt:pkt.version == 0xf1),
+        ConditionalField(XStrFixedLenField("reserved", "", 50), lambda pkt:pkt.version != 0xf1),
+    ]
+
+
+@pytest.fixture(scope="class")
+def higher_retry_count_on_peers(request, nbrhosts):
+    if request.config.getoption("neighbor_type") != "sonic":
+        pytest.skip("Only supported with SONiC neighbor")
+
+    featureCheckResult = nbrhosts[nbrhosts.keys()[0]]['host'].shell(
+            "sudo config portchannel retry-count get PortChannel1", module_ignore_errors=True)
+    if featureCheckResult["rc"] != 0:
+        pytest.skip("SONiC neighbor isn't running supported version of SONiC")
+
+    for nbr in list(nbrhosts.keys()):
+        nbrhosts[nbr]['host'].shell("sudo config portchannel retry-count set PortChannel1 5")
+
+    # Wait for retry count info to be updated
+    time.sleep(5)
+
+    yield
+
+    for nbr in list(nbrhosts.keys()):
+        nbrhosts[nbr]['host'].shell("sudo config portchannel retry-count set PortChannel1 3")
+
+    # Wait for retry count info to be updated
+    time.sleep(90)
+
+
+@pytest.fixture(scope="class")
+def higher_retry_count_on_dut(request, duthost):
+    if request.config.getoption("neighbor_type") != "sonic":
+        pytest.skip("Only supported with SONiC neighbor")
+
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+
+    featureCheckResult = duthost.shell("sudo config portchannel retry-count get {}".format(
+        cfg_facts["PORTCHANNEL"].keys()[0]), module_ignore_errors=True)
+    if featureCheckResult["rc"] != 0:
+        pytest.skip("SONiC DUT isn't running supported version of SONiC")
+
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("sudo config portchannel retry-count set {} 5".format(port_channel))
+
+    # Wait for retry count info to be updated
+    time.sleep(5)
+
+    yield
+
+    for port_channel in list(cfg_facts["PORTCHANNEL"].keys()):
+        duthost.shell("sudo config portchannel retry-count set {} 3".format(port_channel))
+
+    # Wait for retry count info to be updated
+    time.sleep(90)
+
+
+@pytest.fixture(scope="function")
+def config_reload_on_cleanup(request, nbrhosts, duthost):
+    yield
+
+    for nbr in list(nbrhosts.keys()):
+        nbrhosts[nbr]['host'].shell("sudo config reload -y")
+    config_reload(duthost, safe_reload=True)
+
+
+def log_lacpdu_packets(duthost, save_path):
+    """Capture LACPDU packets to file."""
+    # Support for LINUX_SLL2 was added in Scapy 2.5.0. We're using 2.4.5 currently.
+    start_pcap = "tcpdump -i any -y LINUX_SLL -w %s ether proto 0x8809" % (save_path)
+    stop_pcap = "sudo pkill -f '{}'".format(start_pcap)
+    start_pcap = "sudo nohup {} &".format(start_pcap)
+    duthost.shell(start_pcap)
+
+    time.sleep(30)
+
+    duthost.shell(stop_pcap, module_ignore_errors=True)
```

Comment:
`lacpdu.pcap` is not removed from device as part of cleanup. Is that intentional?